staging/src/github.com/kubeedge/beehive/pkg/core/model/message.go

```diff
@@ -53,57 +53,65 @@ type MessageHeader struct {
 	ParentID string `json:"parent_msg_id,omitempty"`
 	// the time of creating
 	Timestamp int64 `json:"timestamp"`
+	// specific resourceVersion for the message, if any.
+	ResourceVersion string `json:"parent_msg_id,omitempty"`
 	// the flag will be set in sendsync
 	Sync bool `json:"sync,omitempty"`
 }
 
-//BuildRouter sets route and resource operation in message
+// BuildRouter sets route and resource operation in message
 func (msg *Message) BuildRouter(source, group, res, opr string) *Message {
```

Comment:
We could plan on changing the function signatures for all set methods. 

Since the receiver's are pointer's of type Message, the functions need not return a pointer.

Reply:
Agreed, let's fixed it in follow up PR